src/lib/rpc.ts

```diff
@@ -108,6 +120,131 @@ export const makeRpcUtils = async ({ agoricNet }) => {
     leader,
     vstorage,
     storageWatcher,
+    agoricNet,
+    netConfigURL,
+    networkConfig,
   };
 };
 export type RpcUtils = Awaited<ReturnType<typeof makeRpcUtils>>;
+
+export enum LoadStatus {
+  Idle = 'idle',
+  Waiting = 'waiting',
+  Received = 'received',
+}
+
+export const usePublishedKeys = (path: string) => {
+  const [status, setStatus] = useState(LoadStatus.Idle);
+  const [data, setData] = useState([]);
+  const rpcUtils = useAtomValue(rpcUtilsAtom);
+
+  useEffect(() => {
+    if (!rpcUtils) {
+      setStatus(LoadStatus.Idle);
+      return;
+    }
+
+    const fetchKeys = async () => {
+      console.debug('usePublishedKeys reading', `published.${path}`);
+      setStatus(LoadStatus.Waiting);
+      const keys = await rpcUtils.vstorage.keys(`published.${path}`);
+      setData(keys);
+      setStatus(LoadStatus.Received);
+    };
+    fetchKeys().catch(e => notifyError(e));
+  }, [path, rpcUtils]);
+
+  return { status, data };
+};
+
+export const usePublishedDatum = (path?: string) => {
+  const [status, setStatus] = useState(LoadStatus.Idle);
+  const [data, setData] = useState({} as any);
+  const rpcUtils = useAtomValue(rpcUtilsAtom);
+
+  useEffect(() => {
+    setData({});
+    if (path === undefined || !rpcUtils) {
+      setStatus(LoadStatus.Idle);
```

Comment:
I see the answer to my question about what `usePublisheDatum(undefined)` does. Very surprising that it triggers an Idle state.

please require a path argument or give a rationale for why that's too difficult.

Reply:
See above comment about `useEffect`